flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1216,6 +1219,10 @@ class ScalarExprLowering {
       llvm::SmallVector<mlir::Attribute> rangeStartIdx;
```

Comment:
Line 1213 uses `array` but `array` is no longer guaranteed to always be initialized. I suspect it is ok there because it has to be a CHARACTER and those, at least now, do not have `genRawVals`. It would be good to add an `assert` above the `do` loop though to verify that `array` is not a `nullptr`.
```c++
assert(array && "array must not be nullptr");
```